src/components/popover/popover.js

```diff
@@ -0,0 +1,210 @@
+
+/**
+ * @ngdoc directive
+ * @name mdPopover
+ * @module material.components.popover
+ * @description
+ * Popovers are used to describe elements that are interactive and either graphical or textual.
+ *
+ * Place a `<md-popover>` as a child of the element it relates to.
+ *
+ * A popover will activate when the user clicks the parent.
+ *
+ * @usage
+ * <hljs lang="html">
+ * <md-icon icon="/img/icons/ic_play_arrow_24px.svg">
+ *   <md-popover>
+ *     Play Music
+ *   </md-popover>
+ * </md-icon>
+ * </hljs>
+ *
+ * @param {expression=} md-visible Boolean bound to whether the popover is currently visible.
+ * @param {expression=} md-placement string bound to location of popover.
+ */
+function MdPopoverDirective($timeout, $window, $$rAF, $document, $mdUtil, $mdTheming, $rootElement) {
+
+  var POPOVER_SHOW_DELAY = 400;
+  var POPOVER_WINDOW_EDGE_SPACE = 8;
+
+  return {
+    restrict: 'E',
+    transclude: true,
+    template:
+      '<div class="md-background"></div>' +
+      '<div class="md-content" ng-transclude></div>',
+    scope: {
+      visible: '=?mdVisible',
+      placement: '=mdPlacement'
+    },
+    link: postLink
+  };
+
+  function postLink(scope, element, attr, contentCtrl) {
+    $mdTheming(element);
+    var parent = element.parent();
+
+    // Look for the nearest parent md-content, stopping at the rootElement.
+    var current = element.parent()[0];
+    while (current && current !== $rootElement[0] && current !== document.body) {
+      if (current.tagName && current.tagName.toLowerCase() == 'md-content') break;
+      current = current.parentNode;
+    }
+    var popoverParent = angular.element(current || document.body);
+
+    // We will re-attach popover when visible
+    element.detach();
+    element.attr('role', 'popover');
+    element.attr('id', attr.id || ('popover_' + $mdUtil.nextUid()));
+
+    parent.on('click', function() {
+        if (scope.visible) setVisible(false);
+        else setVisible(true);
+    });
+
+    scope.$watch('visible', function(isVisible) {
+      if (isVisible) showPopover();
+      else hidePopover();
+    });
+
+    scope.$watch('placement', function (placement) {
+        positionPopover();
+    });
+
+    var debouncedOnResize = $$rAF.debounce(function windowResize() {
+      // Reposition on resize
+      if (scope.visible) positionPopover();
+    });
+    angular.element($window).on('resize', debouncedOnResize);
+
+    // Be sure to completely cleanup the element on destroy
+    scope.$on('$destroy', function() {
+      scope.visible = false;
+      element.remove();
+      angular.element($window).off('resize', debouncedOnResize);
+    });
+
+    // *******
+    // Methods
+    // *******
+
+    // If setting visible to true, debounce to POPOVER_SHOW_DELAY ms
+    // If setting visible to false and no timeout is active, instantly hide the popover.
+    function setVisible(value) {
+      setVisible.value = !!value;
+
+      if (!setVisible.queued) {
```

Comment:
As requested the popover has been created in the style of tooltip.  Perhaps the $mdUtil could be refactored across all components?